pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -367,12 +371,19 @@ func (c *EtcdCertSignerController) populateSecret(secretName, secretNamespace st
 		}
 		return err
 	}
-	secret.Data = map[string][]byte{
-		"tls.crt": cert.Bytes(),
-		"tls.key": key.Bytes(),
+	if err := ensureTLSData(secret); err != nil {
+		secretCopy := secret.DeepCopy()
+		secretCopy.Data = map[string][]byte{
+			"tls.crt": cert.Bytes(),
+			"tls.key": key.Bytes(),
+		}
+		klog.Warningf("secret %s/%s does not have valid data", secretNamespace, secretName)
+		klog.Infof("attempting to update secret %s/%s with valid certs", secretNamespace, secretName)
+		_, err = c.clientset.CoreV1().Secrets(secretNamespace).Update(secretCopy)
```

Comment:
this error would overwrite the original is that intended?

Reply:
yes, I think so. I will log the err getting overridden in line 380

Reply:
done